### PR TITLE
Velocity computed using negative time offset if no valid position at positive offset.

### LINKED ISSUE #4949: Positive time offset doesn't work at stop time
In [EntityView.js#L57](https://github.com/AnalyticalGraphicsInc/cesium/blob/1.30/Source/DataSources/EntityView.js#L57), a small positive time offset of 0.001 seconds is used to get a tiny position delta, used to approximate a velocity near that moment in time.  The time offset does not work for the last 0.001 seconds of playback in the scene, if the objects don't have position data available past the stop time.  This can cause the camera to flip over, for example if it is tracking a GEO satellite and the user moves the timeline to the stop time.

One fix could be to test if the 0.001 offset failed, and attempt a -0.001 offset instead, negating the resulting vector.  I don't think this would be much of a performance hit because it would be a code path only used when the current position is available but a small positive delta position failed.  Can anyone think of other downsides to this possible fix?

### PR DESCRIPTION
**Pull request by klingerj**
Fixes #4949

### CODE REVIEW

**Comment by pjcozzi** [comment]
Welcome @klingerj!!!

**Comment by emackey** [comment]
To test this, load `simple.czml` and track one of the satellites with the camera.  Grab the blue timeline scrubber and pull it all the way to the right, to the end of the timeline.  In master, the camera view goes crazy at the end of the timeline (within one millisecond of the end).  In this branch it should be fixed.

Test coverage of `EntityView` is already poor and unfortunately gets a tiny bit worse with this branch.  The few tests there do pass though.  Perhaps there should be an issue to improve test coverage of this file overall.

**Comment by emackey** [comment]
As predicted, your first merge conflict... :smile:

**Comment by mramato** [comment]
I'm still seeing incorrect camera behavior in this branch.

1. Zoom to ISS in simple.czml
2. Set the time to the slowest multiplier (0.001)
3. Animate backwards.

The camera will still radically change position.

**Comment by emackey** [comment]
Thanks for catching this @mramato!

@klingerj looks like we only tested this with the Molniya satellite, not the ISS.  The Molniya uses a high-Earth "North-Up" view ([these lines](https://github.com/klingerj/cesium/blob/d15c083225cd97e88d781d31364ad98113543235/Source/DataSources/EntityView.js#L99-L119)), but the ISS and GeoEye use the VVLH view ([these lines](https://github.com/klingerj/cesium/blob/d15c083225cd97e88d781d31364ad98113543235/Source/DataSources/EntityView.js#L121-L143)).

In the latter case, the `yBasis` vector comes out negative from what it should be.

We should also test with trucks (Vehicle.czml) and facilities, although I don't expect any problems there due to relative lack of velocity.

File: Source/DataSources/EntityView.js [hunk 1]
```diff
@@ -73,7 +82,11 @@ define([
                     var inertialCartesian = Matrix3.multiplyByVector(toInertial, cartesian, updateTransformCartesian3Scratch5);
                     var inertialDeltaCartesian = Matrix3.multiplyByVector(toInertialDelta, deltaCartesian, updateTransformCartesian3Scratch6);
 
-                    Cartesian3.subtract(inertialCartesian, inertialDeltaCartesian, updateTransformCartesian3Scratch4);
+                    if (invertVelocity) {
+                        Cartesian3.subtract(inertialDeltaCartesian, inertialCartesian, updateTransformCartesian3Scratch4);
```
Comment (emackey) [inline Source/DataSources/EntityView.js]: Looking at this more closely, the result here gets fed straight into `Cartesian3.magnitude` and not used again.  So I don't think the `invertVelocity` flag needs to be checked here at all, as magnitude yields an absolute value.  The flag does need to be checked in the VVLH section below for `yBasis`.

**Comment by mramato** [comment]
Thanks @klingerj!  There was a CHANGES conflict but I took care of it.

**Pull request merged**